common-cli/src/test/java/co/cask/common/cli/CommandSetTest.java

```diff
@@ -108,6 +108,31 @@ private void assertThrows(Callable<Void> callable, Function<Exception, Void> exc
   }
 
   @Test
+  public void testFindMatchWithSpaces() throws Exception {
+    Command greetCommand = new Command() {
+      @Override
+      public void execute(Arguments arguments, PrintStream output) throws Exception {
+        output.println("Hello x!");
+      }
+
+      @Override
+      public String getPattern() {
+        return "greet user x";
+      }
+
+      @Override
+      public String getDescription() {
+        return "Greets x";
+      }
+    };
+
+    CommandSet commandSet = new CommandSet<Command>(ImmutableList.of(greetCommand));
+    CommandMatch match = commandSet.findMatch("greet      user x");
```

Comment:
Can you also test ``" greet user x"``(space in front), ``"greet user   x"`` (multi space after user) and  ``"greet user x "`` (space after end of command)? (I can't find a way to make multi-space sequences appear in this comment...).